test/configure.js

```diff
@@ -0,0 +1,301 @@
+
+            done();
+        }, data => {
+
+            assert.isArray( data, 'Sugarcoat returns an array.' );
+
+            if ( data.length === 1 ) {
+
+                assert.propertyVal( data[0], 'key', 'sections.title', 'We have sections.title in the error array.' );
+            }
+
+            done();
+        });
+    });
+
+    test( 'Title is set to be required. Title errored out when title and dest were not supplied.', done => {
+
+        var configMissingTitleDest = {
+            sections: [
+                {
+                    files: './test/assert/parseVarCode.css'
+                }
+            ]
+        };
+
+        sugarcoat( configMissingTitleDest )
+        .then( data => {
+
+            done();
+        }, data => {
+
+            assert.isArray( data, 'Sugarcoat returns an array.' );
+
+            for ( var errorObj in data ) {
+
+                var key = data[ errorObj ].key;
+
+                if ( key === 'sections.title' ) {
+
+                    assert.propertyVal( data[ errorObj ], 'key', 'sections.title', 'We have sections.title in the error array.' );
+                    break;
+                }
+            }
+
+            done();
+        });
+    });
+
+    test( 'Title is set to be required. Title errored out when all required options were not supplied.', done => {
+
+        var configMissingTitleFiles = {
+            sections: []
+        };
+
+        sugarcoat( configMissingTitleFiles )
+        .then( data => {
+
+            done();
+        }, data => {
+
+            assert.isArray( data, 'Sugarcoat returns an array.' );
+
+            for ( var errorObj in data ) {
+
+                var key = data[ errorObj ].key;
+
+                if ( key === 'sections.title' ) {
+
+                    assert.propertyVal( data[ errorObj ], 'key', 'sections.title', 'We have sections.title in the error array.' );
+                    break;
+                }
+            }
+
+            done();
+        });
+    });
+
+    test( 'Title is set to be required. Title errored out multiple times when more than one files was not supplied.', done => {
+
+        var configMissingtwoTitles = {
+            settings: {
+                dest: './test/documentation'
+            },
+            sections: [
+                {
+                    files: './test/assert/parseVarCode.css'
+                },
+                {
+                    files: './test/assert/parseVarCode.css'
+                }
+            ]
+        };
+
+        sugarcoat( configMissingtwoTitles )
+        .then( data => {
+
+            done();
+        }, data => {
+
+            assert.isArray( data, 'Sugarcoat returns an array.' );
+
+            for ( var errorObj in data ) {
+
+                var key = data[ errorObj ].key;
+
+                if ( key === 'sections.files' ) {
+
+                    assert.propertyVal( data[ errorObj ], 'key', 'sections.title', 'We have sections.title in the error array.' );
```

Comment:
@vbalcom Why would we test the title though? In the test above, we are looking to test the errors for when multiple section.files are missing. We want to make sure that those errors are in the error object. 

@ryanfitzer About a week or two ago, Valerie and I came to the conclusion that when there is an error we want the user to only get those errors. Not the entire config, since they shouldn't be able to do anything with the config object when there are errors present.